src/main/java/com/uber/cadence/internal/sync/TestActivityEnvironmentInternal.java

```diff
@@ -173,7 +174,8 @@ public void registerActivitiesImplementations(Object... activityImplementations)
       task.setActivityType(new ActivityType().setName(activityType));
       IWorkflowService service = new WorkflowServiceWrapper(null);
       Result taskResult =
-          activityTaskHandler.handle(service, testEnvironmentOptions.getDomain(), task);
+          activityTaskHandler.handle(
+              service, testEnvironmentOptions.getDomain(), task, NoopScope.getInstance());
```

Comment:
I would add Metric scope to TestEnvironmentOptions. As users want to instrument their code as well they have to be able to inject their own implementation of the Scope for testing.

Reply:
Good suggestion. I'll do it in the next changeset where I expose the metrics object in Workflow class.